lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1349,6 +1332,32 @@ def determine_group_level_monitored_resource_and_labels(tag,
       [resource, common_labels]
     end
 
+    # Take a locally unique resource id and convert it to the globel unique
+    # monitored resource.
+    def convert_local_resource_id_to_monitored_resource(local_resource_id)
+      return unless local_resource_id
+      if @enable_metadata_agent
+        @log.debug 'Calling metadata agent with local_resource_id: ' \
+                  "#{local_resource_id}."
+        resource = query_metadata_agent_for_monitored_resource(
+          local_resource_id)
+        @log.debug 'Retrieved monitored resource from metadata agent: ' \
+                  "#{resource.inspect}."
+        if resource
+          # TODO(qingling128): Fix this temporary renaming from 'gke_container'
+          # to 'container'.
+          resource.type = 'container' if resource.type == 'gke_container'
```

Comment:
This is actually no longer needed — the logging backend will now happily accept either "container" or "gke_container". May want to call this out as a separate change, though.

Reply:
Wow. Finally. Glad to hear that. Will do it in a separate PR.